plugin/forward/proxy.go

```diff
@@ -48,9 +48,8 @@ func (p *Proxy) Addr() (addr string) { return p.addr }
 func dnsClient(tlsConfig *tls.Config) *dns.Client {
 	c := new(dns.Client)
 	c.Net = "udp"
-	// TODO(miek): this should be half of hcDuration?
-	c.ReadTimeout = 1 * time.Second
-	c.WriteTimeout = 1 * time.Second
+	c.ReadTimeout = hcInterval / 2
+	c.WriteTimeout = hcInterval / 2
```

Comment:
not sure about this change, 250ms may be too few for healthcheck. @miekg are you ok with this change?

Reply:
everything is now getting dynamic, except this, we might be making it too low when something is farther away. 
Can we drop the /2 here?

In the general the hc also also needs to be more complex and do an exp. backoff  - but that's unrelated to this PR

Reply:
I believe it doesn't make sense to allow a longer timeout than the interval. I'll revert this change and send another PR.

Reply:
Actually in case if upstream is down we poll the upstream each `hcInterval + readTimeout` (or `hcInterval + dialTimeout` for TCP), i.e. the time between subsequent polls is `hcInterval + readTimeout`. We could consider decreasing `hcInterval` and increasing `c.ReadTimeout` if we want to tune this logic further

Reply:
alternatively we can consider using ticker in Probe instead of sleep
https://github.com/coredns/coredns/blob/e3534205c7c3a89b09c9a9e2aca1a2159eb9ecde/plugin/pkg/up/up.go#L41

Reply:
Oh, I find it surprising to see that the hcInterval constant only describes a part of the actual interval. Anyway, I reverted this change to the health timeouts and will open another PR.